src/qcodes/parameters/multi_channel_instrument_parameter.py

```diff
@@ -45,16 +48,35 @@ def get_raw(self) -> tuple[ParamRawDataType, ...]:
         """
         return tuple(chan.parameters[self._param_name].get() for chan in self._channels)
 
-    def set_raw(self, value: ParamRawDataType) -> None:
+    def set_raw(self, value: ParamRawDataType | Sequence[ParamRawDataType]) -> None:
         """
-        Set all parameters to this value.
+        Set all parameters to this/these value(s).
 
         Args:
-            value: The value to set to. The type is given by the
+            value: The value(s) to set to. The type is given by the
                 underlying parameter.
         """
-        for chan in self._channels:
-            getattr(chan, self._param_name).set(value)
+        try:
+            for chan in self._channels:
+                getattr(chan, self._param_name).set(value)
+        except Exception as err:
+            try:
+                # Catch wrong length of value before any setting is done
+                value_list = list(value)
+                if len(value_list) != len(self._channels):
+                    raise ValueError
+                for chan, val in zip(self._channels, value_list):
+                    getattr(chan, self._param_name).set(val)
```

Comment:
Would it be possible to add a test that covers this behaviour ?

Reply:
Done.